lib/rule-context.js

```diff
@@ -72,42 +72,27 @@ var PASSTHROUGHS = [
  * @param {object} ecmaFeatures The ecmaFeatures settings passed from the config file.
  */
 function RuleContext(ruleId, eslint, severity, options, settings, ecmaFeatures) {
+    // public.
+    this.id = ruleId;
+    this.options = options;
+    this.settings = settings;
+    this.ecmaFeatures = ecmaFeatures;
+
+    // private.
+    this.eslint = eslint;
+    this.severity = severity;
```

Comment:
Agree with @lo1tuma , we need to keep his these work the same. In my tests, you can speed things up by defining the property first using `this.foo =` and then using `Object.defineProperty`.